### DISCUSSION TITLE
batch mode / command line mode

### DISCUSSION THREAD

**Discussion by masa300V** [asciidocfx/AsciidocFX#604] [Ideas]
Hi, many time I have cases where I want to fix just one or two typo by reading my resulting HTML or PDF, after closing the AsciidocFX window.
I think it's nice to have a command line batch mode to re-generate HTML/PDF from the adoc files **without GUI**.
A request for a command line option like this:

```
AsciidocFX.exe  --batch-mode-PDF  filename.adoc 
```

**Comment by rahmanusta** [comment]
You can try it now, just redownload

https://github.com/asciidocfx/AsciidocFX/releases/tag/v1.8.6


> NOTE: Path to file should be an absolute path

**Windows**

```
# Converts and quit in headless mode (No UI will be visible)
AsciidocFX.exe /path/README.adoc -b pdf -H

# Converts in keeps AsciidocFX running (UI will be visible)
AsciidocFX.exe /path/README.adoc -b pdf -K

# Other output formats
AsciidocFX.exe /path/README.adoc -b pdf
AsciidocFX.exe /path/README.adoc -b html
AsciidocFX.exe /path/README.adoc -b epub
AsciidocFX.exe /path/README.adoc -b docbook
```

**Linux**

Same way as in the Windows


**Mac OS X**

```
# Converts and quit in headless mode (No UI will be visible)
open -W /Applications/AsciidocFX/AsciidocFX.app --args /path/README.adoc -b pdf -H

# Converts in keeps AsciidocFX running (UI will be visible)
open -W /Applications/AsciidocFX/AsciidocFX.app --args /path/README.adoc -b pdf -K

# Other output formats
open -W /Applications/AsciidocFX/AsciidocFX.app --args /path/README.adoc -b pdf
open -W /Applications/AsciidocFX/AsciidocFX.app --args /path/README.adoc -b html
open -W /Applications/AsciidocFX/AsciidocFX.app --args /path/README.adoc -b epub
open -W /Applications/AsciidocFX/AsciidocFX.app --args /path/README.adoc -b docbook
```

**Comment by masa300V** [comment]
@rahmanusta 

Thanks!
I think my Windows PC has difficulties to make it work.
Here, I'm reporting what I can see.

* From cmd.exe, my command goes to background immediately.
Then, this background AsciidocFX.exe stays forever, becomes a zombie.
* From git-bash.exe, my command stays foreground and never finishes.
* I doubt "white spaces" in my path and I moved my adoc file to another path of no white-space still the same.
* My test file is aa.adoc and it contains only one line of "aa" (2 characters)...this source adoc file shouldn't be a reason.
* I can't recall what I did but only one time while I was trying many different ways, I got my aa.pdf result after killing the AsciidocFX.exe...but I can't reproduce it.

**Comment by rahmanusta** [comment]
Can you put what commands did you try?

**Comment by masa300V** [comment]
I tried this
```
(my-path)/AsciidocFX.exe aa.adoc -b pdf -H
```
where "aa.adoc" is just one line of text "aa", nothing asciidoc, just TXT file.

I tried it from both directories where AsciidocFX resides and where aa.adoc resides.
Also I tried both AsciidocFX and aa.adoc of absolute-pathes.

**Comment by masa300V** [comment]
An update:

* I checked Linux version on my WSL Ubuntu.
With "-K" option, I see that an AsciidocFX instance starts at the program folder (AsciidocFX) as the AsciidocFX script brings cwd there.
And any relative-path specification of my adoc file is wrong, the program is looking for the file...this looks like freezing program.
* From this Linux test run, I got a good idea that this CLI mode takes fairly long time to complete.

With above findings, I get back to Windows version ".exe" file.
* I specify the absolute path for my adoc file.
* With the "-K" option, I see the program can find my adoc.
* With the "-H" option, with generous amount of time, I can get my output PDF.

***

* It is working.
* Most probably, we must specify an absolute path of the adoc file.
* It takes fairly long time and need to be patient.
* For debugging, the "-K" option is very helpful.
